exporter/kafkaexporter/kafka_exporter.go

```diff
@@ -297,3 +301,34 @@ func makeSaramaMessages(messages []marshaler.Message, topic string) []*sarama.Pr
 	}
 	return saramaMessages
 }
+
+func messagesWithHeaders(msg []*sarama.ProducerMessage, h []sarama.RecordHeader) {
+	if len(h) == 0 || len(msg) == 0 {
+		return
+	}
+	for i := range msg {
+		if len(msg[i].Headers) == 0 {
+			msg[i].Headers = h
+			continue
+		}
```

Comment:
I would suggest inverting the condition here so you don't need to call continue

Reply:
This simply avoids calling `append()` on a slice that may be nil/empty, and simply setting the headers to the slice that is passed.